deps/wazuh_testing/wazuh_testing/scripts/monitoring.py

```diff
@@ -0,0 +1,261 @@
+
+        time.sleep(options.sleep_time)
+
+
+def get_script_arguments():
+    parser = argparse.ArgumentParser(usage="%(prog)s [options]", description="Wazuh monitoring script",
+                                     formatter_class=argparse.RawTextHelpFormatter)
+    parser.add_argument('-t', '--time', dest='monitoring_time', type=float, required=True, action='store',
+                        help='Type the time in that script will be running.')
+    parser.add_argument('-o', '--output', dest='output_file', required=False, type=str, action='store',
+                        default=None, help='Type the output file name.')
+    parser.add_argument('-s', '--sleep', dest='sleep_time', type=float, default=1, action='store',
+                        help='Type the time in seconds between each entry.')
+    parser.add_argument('-p', '--processes', dest='process_list', required=False, type=str, nargs='+', action='store',
+                        default=["wazuh-db", "wazuh-remoted", "wazuh-analysisd"],
+                        help='Type the processes name to monitor separated by whitespace.')
+    parser.add_argument('-u', '--units', dest='data_unit', default='KB', choices=['B', 'KB', 'MB'],
+                        help='Type unit for the bytes-related values. Default bytes.')
+    parser.add_argument('-v', '--version', dest='version', required=True, help='Version of the binaries. Default none.')
+    parser.add_argument('-d', '--debug', dest='debug', action='store_true', help='Enable debug mode.')
+
+    return parser.parse_args()
+
+
+def main(options):
+    global STOP_STATISTICS_MONITORING
+
+    monitoring_start_time = time.strftime("%Y%m%d-%H%M%S")
+    wazuh_version = options.version
+
+    monitoring_evidences_directory = os.path.join(wazuh_version, monitoring_start_time)
+
+    if options.output_file:
+        monitoring_evidences_directory = options.output_file
+
+    if not os.path.exists(monitoring_evidences_directory):
+        os.makedirs(monitoring_evidences_directory)
+
+    global logger
+    logger = setup_logger(os.path.join(monitoring_evidences_directory, "monitoring.log"), options.debug)
+
+    signal.signal(signal.SIGINT, signal_handler)
+
+    # Start metrics monitoring
+    logger.info(f"Starting metrics monitoring for processes: {options.process_list}")
+    metrics_monitoring_process = subprocess.Popen(f"wazuh-metrics -p {' '.join(options.process_list)} "
+                                                  f"--store {monitoring_evidences_directory}",
+                                                  shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+    logger.info(f"Starting metrics monitoring for processes: {options.process_list}")
+    metrics_monitoring_pid = metrics_monitoring_process.pid
+    statistics_monitoring_thread = threading.Thread(target=collect_data,
+                                                    args=(options, monitoring_evidences_directory,))
+    statistics_monitoring_thread.start()
```

Comment:
Same logger for metrics and statistics

Reply:
Done ab997c4a387846b4818a663df82a282b6378e7bb